File: api-tester/src/main/java/com/revenuecat/apitester/java/EntitlementInfoAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.EntitlementInfo;
import com.revenuecat.purchases.OwnershipType;
import com.revenuecat.purchases.PeriodType;
import com.revenuecat.purchases.Store;
import com.revenuecat.purchases.VerificationResult;

import org.json.JSONObject;

import java.util.Date;

@SuppressWarnings({"unused", "deprecation"})
final class EntitlementInfoAPI {
    static void check(final EntitlementInfo entitlementInfo) {
        final String identifier = entitlementInfo.getIdentifier();
        final boolean active = entitlementInfo.isActive();
        final boolean willRenew = entitlementInfo.getWillRenew();
        final PeriodType periodType = entitlementInfo.getPeriodType();
        final Date latestPurchaseDate = entitlementInfo.getLatestPurchaseDate();
        final Date originalPurchaseDate = entitlementInfo.getOriginalPurchaseDate();
        final Date expirationDate = entitlementInfo.getExpirationDate();
        final Store store = entitlementInfo.getStore();
        final String productIdentifier = entitlementInfo.getProductIdentifier();
        final String productPlanIdentifier = entitlementInfo.getProductPlanIdentifier();
        final boolean sandbox = entitlementInfo.isSandbox();
        final Date unsubscribeDetectedAt = entitlementInfo.getUnsubscribeDetectedAt();
        final Date billingIssueDetectedAt = entitlementInfo.getBillingIssueDetectedAt();
        final OwnershipType ownershipType = entitlementInfo.getOwnershipType();
        final VerificationResult verification = entitlementInfo.getVerification();
    }

    static void checkConstructor(
            String identifier,
            boolean active,
            boolean willRenew,
            PeriodType periodType,
            Date latestPurchaseDate,
            Date originalPurchaseDate,
            Date expirationDate,
            Store store,
            String productIdentifier,
            String productPlanIdentifier,
            boolean sandbox,
            Date unsubscribeDetectedAt,
            Date billingIssueDetectedAt,
            OwnershipType ownershipType,
            JSONObject jsonObject,
            VerificationResult verification
    ) {
        final EntitlementInfo entitlementInfo = new EntitlementInfo(identifier, active, willRenew, periodType,
                latestPurchaseDate, originalPurchaseDate, expirationDate, store, productIdentifier,
                productPlanIdentifier, sandbox, unsubscribeDetectedAt, billingIssueDetectedAt, ownershipType,
                jsonObject,
                verification
        );
        final EntitlementInfo entitlementInfo2 = new EntitlementInfo(identifier, active, willRenew, periodType,
                latestPurchaseDate, originalPurchaseDate, expirationDate, store, productIdentifier,
                productPlanIdentifier, sandbox, unsubscribeDetectedAt, billingIssueDetectedAt, ownershipType,
                jsonObject);
    }

    static void store(final Store store) {
        switch (store) {
            case APP_STORE:
            case MAC_APP_STORE:
            case PLAY_STORE:
            case STRIPE:
            case PROMOTIONAL:
            case UNKNOWN_STORE:
            case AMAZON:
        }
    }

    static void periodType(final PeriodType type) {
        switch (type) {
            case NORMAL:
            case INTRO:
            case TRIAL:
        }
    }

    static void ownershipType(final OwnershipType type) {
        switch (type) {
            case PURCHASED:
            case FAMILY_SHARED:
            case UNKNOWN:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/EntitlementInfoAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.EntitlementInfo
import com.revenuecat.purchases.OwnershipType
import com.revenuecat.purchases.PeriodType
import com.revenuecat.purchases.Store
import org.json.JSONObject
import java.util.Date

@Suppress("unused", "UNUSED_VARIABLE", "DEPRECATION", "LongParameterList")
private class EntitlementInfoAPI {
    fun check(entitlementInfo: EntitlementInfo) {
        with(entitlementInfo) {
            val identifier: String = identifier
            val active: Boolean = isActive
            val willRenew: Boolean = willRenew
            val periodType: PeriodType = periodType
            val latestPurchaseDate: Date = latestPurchaseDate
            val originalPurchaseDate: Date = originalPurchaseDate
            val expirationDate: Date? = expirationDate
            val store: Store = store
            val productIdentifier: String = productIdentifier
            val productPlanIdentifier: String? = productPlanIdentifier
            val sandbox: Boolean = isSandbox
            val unsubscribeDetectedAt: Date? = unsubscribeDetectedAt
            val billingIssueDetectedAt: Date? = billingIssueDetectedAt
            val ownershipType: OwnershipType = ownershipType
            // Trusted entitlements: Commented out until ready to be made public
            // val verification: VerificationResult = verification
        }
    }

    fun checkConstructor(
        identifier: String,
        isActive: Boolean,
        willRenew: Boolean,
        periodType: PeriodType,
        latestPurchaseDate: Date,
        originalPurchaseDate: Date,
        expirationDate: Date?,
        store: Store,
        productIdentifier: String,
        productPlanIdentifier: String?,
        isSandbox: Boolean,
        unsubscribeDetectedAt: Date?,
        billingIssueDetectedAt: Date?,
        ownershipType: OwnershipType,
        jsonObject: JSONObject,
        // Trusted entitlements: Commented out until ready to be made public
        // verification: VerificationResult
    ) {
        val entitlementInfo = EntitlementInfo(
            identifier = identifier,
            isActive = isActive,
            willRenew = willRenew,
            periodType = periodType,
            latestPurchaseDate = latestPurchaseDate,
            originalPurchaseDate = originalPurchaseDate,
            expirationDate = expirationDate,
            store = store,
            productIdentifier = productIdentifier,
            productPlanIdentifier = productPlanIdentifier,
            isSandbox = isSandbox,
            unsubscribeDetectedAt = unsubscribeDetectedAt,
            billingIssueDetectedAt = billingIssueDetectedAt,
            ownershipType = ownershipType,
            jsonObject = jsonObject,
            // Trusted entitlements: Commented out until ready to be made public
            // verification = verification
        )
        val entitlementInfo2 = EntitlementInfo(
            identifier = identifier,
            isActive = isActive,
            willRenew = willRenew,
            periodType = periodType,
            latestPurchaseDate = latestPurchaseDate,
            originalPurchaseDate = originalPurchaseDate,
            expirationDate = expirationDate,
            store = store,
            productIdentifier = productIdentifier,
            productPlanIdentifier = productPlanIdentifier,
            isSandbox = isSandbox,
            unsubscribeDetectedAt = unsubscribeDetectedAt,
            billingIssueDetectedAt = billingIssueDetectedAt,
            ownershipType = ownershipType,
            jsonObject = jsonObject,
        )
    }

    fun store(store: Store) {
        when (store) {
            Store.APP_STORE,
            Store.MAC_APP_STORE,
            Store.PLAY_STORE,
            Store.STRIPE,
            Store.PROMOTIONAL,
            Store.UNKNOWN_STORE,
            Store.AMAZON,
            -> {}
        }.exhaustive
    }

    fun periodType(type: PeriodType) {
        when (type) {
            PeriodType.NORMAL,
            PeriodType.INTRO,
            PeriodType.TRIAL,
            -> {}
        }.exhaustive
    }

    fun ownershipType(type: OwnershipType) {
        when (type) {
            OwnershipType.PURCHASED,
            OwnershipType.FAMILY_SHARED,
            OwnershipType.UNKNOWN,
            -> {}
        }.exhaustive
    }
}
